tests/time_integration/bdf_time_derivative_01.cc

```diff
@@ -0,0 +1,119 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#include <iostream>
+#include <numeric>
+
+#include <deal.II/lac/la_vector.h>
+
+#include <exadg/time_integration/bdf_constants.h>
+
+// Check BDF time derivative given times and quantities
+
+using namespace ExaDG;
+using VectorType = dealii::LinearAlgebra::Vector<double>;
+
+void
+test_constant_quantites(unsigned int const order)
+{
+  std::cout << "Check derivative of constant quantitites" << std::endl;
+
+  // construct constant qunatities
+  std::vector<VectorType> constant_quantities_np_values;
+  for(unsigned int i = 0; i < order + 1; ++i)
+  {
+    VectorType vec(1);
+    vec[0] = 1.0;
+    constant_quantities_np_values.push_back(vec);
+  }
+
+  std::vector<VectorType const *> constant_quantities_np(constant_quantities_np_values.size());
+  std::transform(constant_quantities_np_values.begin(),
+                 constant_quantities_np_values.end(),
+                 constant_quantities_np.begin(),
+                 [](VectorType const & t) { return &t; });
+
+
+  // construct times
+  std::vector<double> times_np(order + 1);
+  std::iota(times_np.begin(), times_np.end(), 0.0);
+  std::sort(times_np.begin(), times_np.end(), [](auto const a, auto const b) { return a > b; });
+
+  // compute temporal derivative
+  VectorType derivative(1);
+  derivative[0] = -99.0;
+
+  compute_bdf_time_derivative(derivative, constant_quantities_np, times_np);
+
+  // check if derivative is 0
+  assert(derivative[0] < 1e-12);
+  std::cout << "OK" << std::endl;
+}
+
+void
+test_linear_changing_quantites(unsigned int const order)
+{
+  std::cout << "Check derivative of linear changing quantitites" << std::endl;
+
+  // construct qunatities with slope -1
```

Comment:
-> quantities

Now, we have all I think :)

Reply:
[e7c9fe3](https://github.com/exadg/exadg/pull/458/commits/e7c9fe31d7f63ff8ab425c4b392fd36f4dd4f637)